tools/check-symbol-names

```diff
@@ -20,11 +20,19 @@ set -o pipefail
 
 nm -A -P "$@" | awk '
 # ignore caml_foo, camlFoo_bar, _caml_foo, _camlFoo_bar
-$2 ~ /^(_?caml[_A-Z])/ { next }
+$2 ~ /^(_?[c|C]aml[_A-Z])/ { next }
```

Comment:
Out of curiosity, where are the `Caml` / `_Caml` symbols from? (Names like `_Caml` are verboten in C so we should probably change any we're using).

Reply:
I think the underscore comes from the translation from C symbol to asm symbol.

As for `Caml`, the only occurrence is `Caml_state`, an unfortunate deviation from our usual convention, but it's too late in the release cycle to fix it before 4.10 (and it's not a major problem anyway).

We could change this code to accept only `Caml_state` (instead of `Caml_*`) and decide later whether we want to risk renaming the variable to `caml_state`.

Reply:
That's the Windows encoding of `Caml_state`, I expect

Reply:
Shouldn't it be `[cC]` or `\(c\|C\)`?